### PR TITLE
Multi monitor layout

### PR DESCRIPTION
**Pull request by lassoan**
Layouts can now contain multiple viewports. The default viewport is displayed as the center widget of the application main window, additional Additional viewports can be independent top-level windows or docking windows.

Views can be maximized/restored separately in each viewport.

Currently, a single new dual-viewport layout is added: Dual Monitor Four-Up View

image

Associated CTK pull request (already merged): https://github.com/commontk/CTK/pull/1061 first.

### CODE REVIEW

**Comment by jamesobutler** [comment]
Very cool! Is this aiming to address long time issue https://github.com/Slicer/Slicer/issues/2104?

**Comment by lassoan** [comment]
> Very cool! Is this aiming to address long time issue #2104?

Yes!

**Review by pieper — APPROVED** [review]
Nice!

File: Libs/MRML/Widgets/qMRMLLayoutManager.h [hunk 1]
```diff
@@ -167,7 +167,7 @@ class QMRML_WIDGETS_EXPORT qMRMLLayoutManager : public ctkLayoutFactory
   int layout()const;
 
   /// Return the view node that is temporarily shown maximized in the view layout.
-  Q_INVOKABLE vtkMRMLAbstractViewNode* maximizedViewNode();
+  //Q_INVOKABLE vtkMRMLAbstractViewNode* maximizedViewNode();
```
Comment (cpinter) [inline Libs/MRML/Widgets/qMRMLLayoutManager.h:170]: You could remove this I guess.

**Comment by cpinter** [comment]
This is great, thanks a lot @lassoan ! I reviewed the code and have one thing I don't understand (just curiosity): who calls `createViewport`? I cannot seem to find it in the diff (and there were no collapsed sections so have no idea)

File: Libs/MRML/Widgets/qMRMLLayoutManager.h (hunk 1 above)
Comment (lassoan) [reply Libs/MRML/Widgets/qMRMLLayoutManager.h]: done

**Comment by lassoan** [comment]
> This is great, thanks a lot @lassoan ! I reviewed the code and have one thing I don't understand (just curiosity): who calls `createViewport`? I cannot seem to find it in the diff (and there were no collapsed sections so have no idea)

It is a virtual function called by the base class. I've added a comment into the method to clarify this.

**Comment by lassoan** [comment]
Thanks for the reviews. The associated CTK changes have been merged, so I'll merge this now.

**Pull request merged**